Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -4844,6 +4846,15 @@ LayoutRect RenderLayer::localBoundingBox(OptionSet<CalculateLayerBoundsFlag> fla
             box->flipForWritingMode(result); // The mask clip rect is in physical coordinates, so we have to flip, since localBoundingBox is not.
         } else
             result = box->visualOverflowRect();
+
+        if (flags & IncludeRootBackgroundPaintingArea && renderer().isDocumentElementRenderer()) {
+            // If the root layer becomes composited (e.g. because some descendant with negative z-index is composited),
+            // then it has to be big enough to cover the viewport in order to display the background. This is akin
+            // to the code in RenderBox::paintRootBoxFillLayers().
+            const LocalFrameView& frameView = renderer().view().frameView();
+            result.setWidth(std::max(result.width(), frameView.contentsWidth() - result.x()));
+            result.setHeight(std::max(result.height(), frameView.contentsHeight() - result.y()));
```

Comment:
Do we ever end up making a transparency layer with this size? Could it become huge, or will we still clip to tile boundaries?

Reply:
The transparency layer is restricted to the clip rect on the context, which includes the bounds of the context.